src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs

```diff
@@ -821,6 +883,18 @@ protected override void OnGotFocus(EventArgs e)
             Invalidate();
         }
 
+        // Called by the ICommandProvider's internal DIM-based logic.
+        void ICommandProvider.HandleCommandChanged(EventArgs e)
+            => OnCommandChanged(e);
+
+        // Called by the ICommandProvider's internal DIM-based logic.
+        void ICommandProvider.HandleCommandCanExecuteChanged(object sender, CommandEventArgs e)
```

Comment:
To align with .NET runtime annotations:
```suggestion
        void ICommandProvider.HandleCommandCanExecuteChanged(object? sender, CommandEventArgs e)
```

ditto for other instances.